command/harness/destroy.go

```diff
@@ -38,13 +38,16 @@ func HandleDestroy(ctx context.Context, r *VMCleanupRequest, s store.StageOwnerS
 		WithField("stage_runtime_id", r.StageRuntimeID).
 		WithField("pool", r.PoolID)
 
-	if err := poolManager.Destroy(ctx, r.PoolID, inst.ID); err != nil {
+	if err = poolManager.Destroy(ctx, r.PoolID, inst.ID); err != nil {
 		return fmt.Errorf("annot destroy the instance: %w", err)
 	}
 	logr.Traceln("destroyed instance")
-
-	if err := s.Delete(ctx, r.StageRuntimeID); err != nil {
-		logrus.WithError(err).Errorln("failed to delete stage owner entity")
+	exists, _ := s.Find(ctx, r.StageRuntimeID, r.PoolID)
+	if exists != nil {
+		if deleteErr := s.Delete(ctx, r.StageRuntimeID); err != nil {
+			logrus.WithError(deleteErr).Errorln("failed to delete stage owner entity")
```

Comment:
Use logr here since it has some other fields populated

Reply:
fixed